pkgs/development/compilers/cudatoolkit/common.nix

```diff
@@ -214,7 +214,6 @@ stdenv.mkDerivation rec {
     description = "A compiler for NVIDIA GPUs, math libraries, and tools";
     homepage = "https://developer.nvidia.com/cuda-toolkit";
     platforms = [ "x86_64-linux" ];
-    license = licenses.unfree;
```

Comment:
Are your sure that the SDK by itself can re-distributed? The EULA states "The following CUDA Toolkit files may be distributed with Licensee **Applications** developed by you, [...]" (Sec. 2.6).

Reply:
See section `2.3`

```
2.3. Operating Systems

Those portions of the SDK designed exclusively for use on the Linux or FreeBSD operating systems,
or other operating systems derived from the source code to these operating systems,
may be copied and redistributed for use in accordance with this Agreement,
provided that the object code files are not modified in any way (except for unzipping of compressed files).
```

Reply:
Looks like Conda distributes the `.so` files only for CDUA. You're correct that we cannot distribute the entire SDK, I believe we'll need to make a separate derivation with the redistributable components, so I suppose there's an argument for keeping this derivation as unfree and a new derivation under the `nvidiaCuda` license

Reply:
> provided that the object code files are not modified in any way
So patching the RPATHs doesn't sound possible under this agreement.

Reply:
Also the question is if that is not just metadata as the code is unmodified. You are also allowed to change the file creation time.

Reply:
Maybe we can ask NVIDIA for an exemption/addition.

Reply:
I think it’s possible to get working without modifying binary by wrapping LD_LIBRARY_PATH: https://github.com/NixOS/nixpkgs/issues/76512

Reply:
Is there a precedence on whether modifying library metadata such as the dynamic section (i.e. `DT_RUNPATH`) constitutes a modification of the object code? I've tried finding an answer but couldn't find anything.

Reply:
Good idea, xbreak!

Reply:
@tbenst did you hear anything back?

Reply:
@danieldk, no, but here’s the email trail: https://github.com/NixOS/nixpkgs/pull/76233#discussion_r481478321